src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+
+	for (iteration_counter = 1; iteration_counter <= max_iterations; iteration_counter++)
+	{
+		//	wxPrintf("Starting iteration number %li\n\n", iteration_counter);
+		max_shift = -FLT_MAX;
+		max_rotation = -FLT_MAX;
+
+		float average_image_psi_rotation = 0.;
+		float average_image_x_shift = 0.;
+		float average_image_y_shift = 0.;
+
+		// make the current running average if necessary
+
+		if (number_of_frames_for_running_average > 1)
+		{
+			#pragma omp parallel for default(shared) num_threads(max_threads) private(image_counter, start_frame_for_average, end_frame_for_average, running_average_counter)
+			for (image_counter = 0; image_counter < number_of_images; image_counter++)
+			{
+				start_frame_for_average = image_counter - running_average_half_size;
+				end_frame_for_average = image_counter + running_average_half_size;
+
+				if (start_frame_for_average < 0)
+				{
+					end_frame_for_average -= start_frame_for_average; // add it to the right
+					start_frame_for_average = 0;
+				}
+
+				if (end_frame_for_average >= number_of_images)
+				{
+					start_frame_for_average -= (end_frame_for_average - (number_of_images - 1));
+					end_frame_for_average = number_of_images - 1;
+				}
+
+				if (start_frame_for_average < 0)
+					start_frame_for_average = 0;
+				if (end_frame_for_average >= number_of_images)
+					end_frame_for_average = number_of_images - 1;
+				running_average_stack[image_counter].SetToConstant(0.0f);
+
+				for (running_average_counter = start_frame_for_average; running_average_counter <= end_frame_for_average; running_average_counter++)
+				{
+					running_average_stack[image_counter].AddImage(&input_stack_times_ctf[running_average_counter]);
+				}
+			}
+		}
+
+		// do no subtract current image from sum and do not shift in y-direction
+		#pragma omp parallel default(shared) num_threads(max_threads) private(image_counter, sum_of_images_temp, current_peak, psi, best_inplane_score, best_inplane_values)
+		{ // for omp
+
+			sum_of_images_temp.Allocate(input_stack_times_ctf[0].logical_x_dimension, input_stack_times_ctf[0].logical_y_dimension, false);
+
+		#pragma omp for
+		for (image_counter = 0; image_counter < number_of_images; image_counter++)
+		{
+			/*	
+			// prepare the sum reference by subtracting out the current image, applying a bfactor and masking central cross
+			sum_of_images_temp.CopyFrom(&sum_of_images);
+			//sum_of_images_temp.SubtractImage(&stack_for_alignment[image_counter]); // turned-off
+			sum_of_images_temp.ApplyBFactor(unitless_bfactor);
+
+			if (mask_central_cross == true)
+			{
+				sum_of_images_temp.MaskCentralCross(width_of_vertical_line, width_of_horizontal_line);
+			}
+
+			sum_of_images_temp.BackwardFFT();
+			// debug by writing out sum_of_images_temp
+			//sum_of_images_temp.QuickAndDirtyWriteSlice("my_rotated_sum.mrc", 1);
+			*/
+					
+			best_inplane_score = - std::numeric_limits<float>::max();
```

Comment:
Yay! I like this use of stl here : ) Maybe make  lines 780/781 consistent with this?